core/src/main/java/org/bitcoinj/script/ScriptChunk.java

```diff
@@ -108,33 +108,37 @@ public boolean isShortestPossiblePushData() {
         return opcode == OP_PUSHDATA4;
     }
 
-    public void write(OutputStream stream) throws IOException {
+    public byte[] toByteArray() {
         if (isOpCode()) {
             checkState(data == null);
-            stream.write(opcode);
+            return new byte[]{ (byte) opcode };
         } else if (data != null) {
+            ArrayList<Byte> res = new ArrayList<>();
             if (opcode < OP_PUSHDATA1) {
                 checkState(data.length == opcode);
-                stream.write(opcode);
+                res.add((byte) opcode);
             } else if (opcode == OP_PUSHDATA1) {
                 checkState(data.length <= 0xFF);
-                stream.write(OP_PUSHDATA1);
-                stream.write(data.length);
+                res.add((byte) OP_PUSHDATA1);
+                res.add((byte) data.length);
             } else if (opcode == OP_PUSHDATA2) {
                 checkState(data.length <= 0xFFFF);
-                stream.write(OP_PUSHDATA2);
-                stream.write(0xFF & data.length);
-                stream.write(0xFF & (data.length >> 8));
+                res.add((byte) OP_PUSHDATA2);
+                res.add((byte) (0xFF & data.length));
+                res.add((byte) (0xFF & (data.length >> 8)));
             } else if (opcode == OP_PUSHDATA4) {
                 checkState(data.length <= Script.MAX_SCRIPT_ELEMENT_SIZE);
-                stream.write(OP_PUSHDATA4);
-                Utils.uint32ToByteStreamLE(data.length, stream);
+                res.add((byte) OP_PUSHDATA4);
+                res.add((byte) (0xFF & data.length));
+                res.add((byte) (0xFF & (data.length >> 8)));
+                res.add((byte) (0xFF & (data.length >> 16)));
+                res.add((byte) (0xFF & (data.length >> 24)));
             } else {
                 throw new RuntimeException("Unimplemented");
             }
-            stream.write(data);
+            return Bytes.concat(Bytes.toArray(res), data);
```

Comment:
Imho this has the potential of a performance regression, as it copies around res and data (multiple times).